### PR TITLE
Setup GitHub actions

### LINKED ISSUE #223: port to github actions
I'd like to move our CI from Travis to Github Actions. It needs to do two things:

* run tests
* build and publish the mdbook

Anybody want to help?

### PR DESCRIPTION
**Pull request by mbrobbel**
This is an attempt to close #223.

### CODE REVIEW

**Comment by nikomatsakis** [comment]
Woohoo, thanks!

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions-rs/toolchain@v1
+      with:
+        toolchain: ${{ matrix.rust }}
+        default: true
+    - uses: actions-rs/cargo@v1
+      with:
+        command: test
+        args: --all-features --all-targets
```
Comment (Marwes) [inline .github/workflows/test.yml:36]: `--all-targets` ends up not testing `src/doctest.rs`. Either this should be removed or multiple commands needs to be run.

**Comment by nikomatsakis** [comment]
Regarding tokens, can we not make use of the [`GITHUB_TOKEN` that is always available](https://help.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token)? I would rather not put a personal access token in there, as it gives access to way more than is needed.

**Comment by mbrobbel** [comment]
Yes, let's try with `GITHUB_TOKEN`. I've had issues before where using the provided `GITHUB_TOKEN` wouldn't trigger a GitHub Pages build, but maybe that is fixed now.

**Comment by mbrobbel** [comment]
I switched to a different action for the deployment to GitHub Pages, which supports using `GITHUB_TOKEN`.

**Comment by nikomatsakis** [comment]
Should I merge, or is there more to be done?

**Comment by mbrobbel** [comment]
It should be good to go. If there are issues I can follow up with another PR.

**Pull request merged**